digid_eherkenning/oidc/models/digid.py

```diff
@@ -0,0 +1,83 @@
+from collections.abc import Collection
+
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from django_jsonform.models.fields import ArrayField
+from mozilla_django_oidc_db.typing import ClaimPath
+
+from .base import OpenIDConnectBaseConfig, get_default_scopes_bsn
+
+
+class DigiDConfig(OpenIDConnectBaseConfig):
+    """
+    Configuration for DigiD authentication via OpenID connect
+    """
+
+    identifier_claim_name = models.CharField(
+        _("BSN claim name"),
+        max_length=100,
+        help_text=_("The name of the claim in which the BSN of the user is stored"),
+        default="bsn",
+    )
+    oidc_rp_scopes_list = ArrayField(
+        verbose_name=_("OpenID Connect scopes"),
+        base_field=models.CharField(_("OpenID Connect scope"), max_length=50),
+        default=get_default_scopes_bsn,
+        blank=True,
+        help_text=_(
+            "OpenID Connect scopes that are requested during login. "
+            "These scopes are hardcoded and must be supported by the identity provider"
+        ),
+    )
+
+    class Meta:
+        verbose_name = _("OpenID Connect configuration for DigiD")
+
+    @property
+    def oidcdb_username_claim(self) -> list[str]:
```

Comment:
Why do we return a list here?

Reply:
we use the bits of a claim to specify the nesting, so that we can support <kbd>.</kbd> characters inside a claim. I should update the type hint to use `ClaimPath` :)